django/contrib/sessions/middleware.py

```diff
@@ -57,10 +57,10 @@ def process_response(self, request, response):
                         try:
                             request.session.save()
                         except UpdateError:
-                            # The user is now logged out; redirecting to same
-                            # page will result in a redirect to the login page
-                            # if required.
-                            return redirect(request.path)
+                            # Session has been deleted (the user is logged out or so).
+                            # Raising SuspiciousOperation exception indicating that
+                            # request can not be properly performed.
+                            raise SuspiciousOperation('Session already destroyed, request failed')
```

Comment:
I'm not sure if the message is descriptive enough that a developer will understand if this is some bug in their app or if it's something they can safely ignore.